include/circt/Dialect/RTL/Types.td

```diff
@@ -81,12 +81,28 @@ class FieldInfo_Type<string name> : RTLType<name> {
   }];
 }
 
-def StructType : FieldInfo_Type<"Struct"> {
+def StructType : FieldInfo_Type<"Struct">, Type<CPred<"$_self.isa<StructType>()">, "StructType"> {
```

Comment:
With this commit https://github.com/llvm/llvm-project/commit/95019de8a122619fc038c9fe3c80e625e3456bbf we shouldn't need to inherit from type anymore.

Reply:
nice

Reply:
We have this now that we updated the LLVM submodule.